src/main.rs

```diff
@@ -396,29 +396,27 @@ fn run() -> Result<(), failure::Error> {
     } else if let Some(matches) = matches.subcommand_matches("generate") {
         let name = matches.value_of("name").unwrap_or("worker");
         let site = matches.is_present("site");
+        let template = matches.value_of("template");
+        let mut target_type = None;
 
-        let (target_type, template) = if site {
-            // Workers Sites projects are always Webpack for now
-            let target_type = Some(TargetType::Webpack);
-            let template = "https://github.com/cloudflare/worker-sites-template";
-
-            (target_type, template)
+        let template = if site {
+            if template.is_some() {
```

Comment:
hesitant to keep this gate but i may be in the minority; folks wanting to write custom site workers should be encouraged, shouldn't they? perhaps rather than gating this we should expose the api? idk.

Reply:
Right now if you try to run this command it is equivalent to `wrangler generate --site` -- if people want to use their own boilerplate instead of ours that's fine, but it won't work with the `--site` flag. (you can try it yourself! the behavior is super wonky)

Reply:
This PR also allows templates to include `[site]` sections in their tomls, so the `--site` flag wouldn't be needed to create that section of the toml if the template was properly configured

Reply:
sure, but is that because `--site` is being too special?

Reply:
yes. it's essentially an alias to a specific template, but that's our API for now. open to looking at ways to fix this but it's not gonna make it in this PR